craft_application/commands/lifecycle.py

```diff
@@ -74,11 +74,17 @@ def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
             nargs="*",
             help="Optional list of parts to process",
         )
-        parser.add_argument(
+        group = parser.add_mutually_exclusive_group()
+        group.add_argument(
             "--destructive-mode",
             action="store_true",
             help="Build in the current host",
         )
+        group.add_argument(
```

Comment:
I'm not sure this is something we want in all `craft-application` apps. Currently both rockcraft and charmcraft default to `lxd` on Linux with no official way (at least in the case of Charmcraft, but I'm 80% sure in the case of rockcraft) to switch to the multipass backend.

I think this can be done (almost?) entirely in snapcraft through official means, though not quite as cleanly as this:

1. Register a [parser filler](https://github.com/canonical/craft-application/blob/4d94565ad2788c11fee7974685bc6af207d3dca6/craft_application/commands/base.py#L112) on `LifecyclePartsCommand` to add `--use-lxd`
2. Register a prologue that checks for `--destructive-mode` and `--use-lxd` both being used and errors out (<-- this is the "less clean" part since we can't put them in a mutex group)
3. In `Snapcraft.pre_run` check for the flag in the parser's arguments and make a change that will allow the ProviderService to use this.

This would need a change to the ProviderService - take `provider` as an optional parameter on `__init__` rather than always setting it to `None` - then `Snapcraft._configure_services()` can set kwarg for the provider service.

Reply:
It seems counterintuitive to me to have a command line argument for the purpose of reverting to the default config. I think I'd prefer something like a `--provider` that can take a craft-providers backend name + `host`, with `--destructive-mode` being a synonym for `--provider=host`.

That would then also work nicely with a `CRAFT_PROVIDER` environment variable and a `provider` snap option (where `--provider` > `CRAFT_PROVIDER` > snap option > default).

Reply:
We do eventually want to support multipass for Charmcraft, we actually do, when running on macos. We should be ok with changing implementation logic, but let's not change UX logic as that has been discussed, for this specific thing, more times than I have fingers for and for more hours than my age number in years